### PR TITLE
feat: close button for banner (#2912)

### LINKED ISSUE #2912: Add close functionality for banner
- [ ] Add close button to the right of the "Hire us" button
- [ ] Make it so it's easy to update and change banner (maybe move all logic to one file or something)
- [ ] Use local storage to store "closed" state for particular banner + you can also use something like "banner: 'official-partner'" in local storage, so that we would just change value for new banners.

https://vuestic.dev/

image

### PR DESCRIPTION
**Pull request by kushich**
## Description
Close icon for banner to hide it from docs and landing. Hided state will be saved in the localStorage.
Screenshot 2023-02-08 at 10 13 14

closes #2912

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/docs/src/components/HeaderBanner.vue [hunk 1]
```diff
@@ -23,13 +23,26 @@
...
+          <va-icon class="cursor-pointer" size="small" name="close" @click="hide" />
+        </div>
       </div>
     </div>
   </div>
 </template>
 
+<script setup lang="ts">
+import { ref } from 'vue'
+const hided = ref(localStorage.getItem('official-partner-banner') === 'hide')
```
Comment (rustem-nasyrov) [inline packages/docs/src/components/HeaderBanner.vue:38]:
```suggestion
const isHidden = ref(localStorage.getItem('official-partner-banner') === 'hide')
```

**Comment by rustem-nasyrov** [comment]
@kushich, could you please resolve conflicts and merge PR?

File: packages/docs/src/components/HeaderBanner.vue [hunk 2]
```diff
@@ -23,13 +23,26 @@
...
+          <va-icon class="cursor-pointer" size="small" name="close" @click="hide" />
+        </div>
       </div>
     </div>
   </div>
 </template>
 
+<script setup lang="ts">
+import { ref } from 'vue'
+const isHidden = ref(localStorage.getItem('official-partner-banner') === 'hide')
```
Comment (asvae) [inline packages/docs/src/components/HeaderBanner.vue:38]:
```suggestion
const isHidden = ref(localStorage.getItem('banner') === 'official-partner')
```

The idea is that only one banner should be available, and we also do not want to keep a key in local storage for each banner historically.

File: packages/docs/src/components/HeaderBanner.vue [hunk 3]
```diff
@@ -23,13 +23,26 @@
...
+
+const hide = () => {
+  localStorage.setItem('official-partner-banner', 'hide')
```
Comment (asvae) [inline packages/docs/src/components/HeaderBanner.vue:41]:
```suggestion
  localStorage.setItem('banner', 'official-partner')
```

**Comment by asvae** [comment]
Could you also add an image to PR, so that it's easier to review?

File: packages/docs/src/components/HeaderBanner.vue (hunk 2 above)
Comment (kushich) [reply packages/docs/src/components/HeaderBanner.vue:38]: got it

**Comment by zvenigorodskaia** [comment]
Close functionality is neccessary only for docs pages, not for Landing
For Landing blue banner should be always visible
For Docs page - should has close Icon 
After click please add animation 300ms cubic-bezier(.4,0,.2,1); for hiding banner and please save this state at the Local storage

**Comment by kushich** [comment]
@zvenigorodskaia 
https://user-images.githubusercontent.com/45711229/218407430-91ae3557-0554-4648-a946-57195f04f04c.mov

**Comment by kushich** [comment]
@zvenigorodskaia 
banner

**Review by m0ksem — APPROVED** [review]

**Comment by m0ksem** [comment]
Will move to a different PR and adapt to nuxt & ssr.

**Pull request closed without merging**